client/shipping-label/views/purchase/steps/packages/list.js

```diff
@@ -1,78 +1,104 @@
 import React, { PropTypes } from 'react';
 import { translate as __ } from 'lib/mixins/i18n';
-import NumberField from 'components/number-field';
-import FormLegend from 'components/forms/form-legend';
-import { sprintf } from 'sprintf-js';
-import _ from 'lodash';
+import Gridicon from 'components/gridicon';
+import classNames from 'classnames';
 
-const renderPackageDimensions = ( pckg, dimensionUnit ) => {
-	return `${pckg.length} ${dimensionUnit} x ${pckg.width} ${dimensionUnit} x ${pckg.height} ${dimensionUnit}`;
-};
+const PackageList = ( { selected, all, unpacked, packageId, openPackage, addPackage } ) => {
+	const renderUnpackedLink = () => {
+		if ( ! unpacked.length ) {
+			return null;
+		}
 
-const OrderPackages = ( { packages, updateWeight, dimensionUnit, weightUnit, errors } ) => {
-	const renderItemInfo = ( item, itemIndex ) => {
 		return (
-			<div key={ itemIndex } className="wcc-package-item">
-				<div className="wcc-package-item__name">
-					<span className="wcc-package-item__title">
-						<a href={ item.url } target="_blank">{ item.name }</a>
-					</span>
-					{ item.attributes && <p>{ item.attributes }</p> }
-				</div>
-				<div className="wcc-package-item__quantity">{ item.quantity }</div>
+			<div className="wcc-packages-list__unpacked">
+				<a href="#" className="wcc-packages-list__link" onClick={ () => ( openPackage( '' ) ) }>{ __( 'Saved for later' ) }</a>
 			</div>
 		);
 	};
 
-	const numPackages = Object.keys( packages ).length;
-	let pckgIndex = 1;
+	const renderAddPackage = () => {
+		const boxesKeys = Object.keys( all );
+		if ( ! boxesKeys.length ) {
+			return null;
+		}
 
-	const renderPackageInfo = ( pckg, pckgId ) => {
-		const pckgErrors = errors[ pckgId ] || {};
-		return (
-			<div key={ pckgId }>
-				<div className="wcc-package-package-number">
-					{ sprintf( __( 'Package %d (of %d)' ), pckgIndex++, numPackages ) }
+		return ( <div className="wcc-packages-list__add-container">
+			<a href="#" className="wcc-packages-list__link" onClick={ () => ( addPackage() ) }>
+				<Gridicon icon="add-outline" size={ 18 } /> { __( 'Add package' ) }
+			</a>
+		</div> );
+	};
+
+	let boxCount = 0;
+	const packed = [];
+	const individual = [];
+
+	Object.keys( selected ).forEach( ( pckgId ) => {
+		const pckg = selected[ pckgId ];
+
+		if ( 'individual' === pckg.box_id ) {
+			individual.push(
+				<div className="wcc-packages-list__item" key={ pckgId }>
```

Comment:
Let's keep this DRY - can you write a render method for package list items? (This is copied almost verbatim below for boxes/envelopes)